### PR TITLE
Fix a bug in older firefox versions

### PR DESCRIPTION
**Pull request by jescalan**
🎟️ [Asana Task]()
🔍 [Preview Link](https://react-components-git-{branch-slug}-hashicorp.vercel.app)

---

## Description

Potentially fixes https://github.com/hashicorp/vault/issues/13350

Tested in saucelabs, breaking down [the shorthand](https://developer.mozilla.org/en-US/docs/Web/CSS/flex) seems to resolve the issue.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

**Comment by dstaley** [comment]
Applying the expanded form doesn't seem to resolve the issue for me in 78esr

image

**Comment by jescalan** [comment]
@dstaley could you try again but with dropping the `flex-basis`? I just adjusted the PR for this 😬

**Comment by dstaley** [comment]
> @dstaley could you try again but with dropping the `flex-basis`? I just adjusted the PR for this 😬

Negative ghost rider.

image

**Comment by jescalan** [comment]
@dstaley ah you also need to disable `flex` declaration attached directly to the selector for it to kick in 😁

**Comment by zchsh** [comment]
Tested on SauceLabs, I think I'm seeing the same thing as @dstaley . When I add `min-width: 0` to `#inner` it seems to resolve the issue though 👍 I think this has to do with how the default `min-width: auto` works in `flex` contexts (https://drafts.csswg.org/css-flexbox/#min-size-auto) but am unclear on the details (have just ran into this before!).

**Comment by zchsh** [comment]
CleanShot 2021-12-07 at 15 22 49@2x

**Comment by zchsh** [comment]
Edit: looked into this a bit more and found a simpler fix. It seems that the defaults for `flex-basis` have changed over time. Explicitly setting  `flex: 1 1 auto` seems to resolve the issue as well.

CleanShot 2021-12-07 at 15 27 53@2x

File: packages/docs-page/style.module.css [hunk 1]
```diff
@@ -104,7 +104,8 @@ which is rendered into docs-sidenav */
   padding: 0 1rem;
 
   @media (--desktop-viewports) {
-    flex: 1;
+    flex-grow: 1;
+    flex-shrink: 1;
```
Comment (zchsh) [inline packages/docs-page/style.module.css:108]: I'm honestly having trouble understanding all the nuance both of flex-basis and how it has changed over the years (https://developer.mozilla.org/en-US/docs/Web/CSS/flex-basis#values). But taking a more pragmatic approach, the CSS below seems to work both in FireFox 78 and in newer browsers:

```suggestion
    flex: 1 1 auto;
```

As an alternate solution, this also seems to work:

```suggestion
    flex: 1 1;
    min-width: 0;
```

**Review by zchsh — CHANGES_REQUESTED** [review]

**Comment by jescalan** [comment]
Sounds good, applied the suggestion!

**Review by zchsh — APPROVED** [review]

**Comment by zchsh** [comment]
💖 ☝️ Sorry for the delay 😬

**Pull request merged**